react/react2/week1/react2week1/src/FetchData.js

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useContext, useState } from "react";
+import { SearchContext } from "./GithubSearch";
+
+export function FetchData() {
+  const UserContext = useContext(SearchContext);
+  const [userfetch, setUserFetch] = useState([]);
+  const [errormessage, setErrorMessage] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (!UserContext) {
+      setUserFetch([]);
+      return;
+    }
+    const API = `https://api.github.com/search/users?q=${UserContext}`;
```

Comment:
Here you quickly get the github API limit. When working with code like this, you should create your own mock backend to test on. Create a simple node backend like the one we used in the shifts assignment with a hardcoded list of mock github users.